tests/testthat/test-gather_fails.R

```diff
@@ -0,0 +1,469 @@
+library(shiny)
+library(shinyvalidate)
+
+
+test_that("invalid arguments raise errors", {
+  expect_error(gather_files("string"))
+  expect_error(gather_files_com("string"))
+  expect_error(gather_files_grp(list("name" = "string")))
+})
+
+
+test_that("gather_fails: valid inputs produce desired output", {
+  server <- function(input, output, session) {
+    iv <- InputValidator$new()
+    iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter"))
+    iv$add_rule("number", ~ if (as.integer(.) %% 2L == 1L) "choose an even number")
+    iv$enable()
+    values <- reactive({
+      gather_fails(iv)
+      list(
+        "letter" = input[["letter"]],
+        "number" = input[["number"]]
+      )
+    })
+  }
+
+  shiny::testServer(server, {
+    session$setInputs(
+      "letter" = "A",
+      "number" = 2L
+    )
+    expect_identical(values(), list(
+      "letter" = input[["letter"]],
+      "number" = input[["number"]]
+    ))
+  })
+})
+
+test_that("gather_fails_com: valid inputs produce desired output", {
+  server <- function(input, output, session) {
+    iv <- InputValidator$new()
+    iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter"))
+    iv$add_rule("number", ~ if (as.integer(.) %% 2L == 1L) "choose an even number")
+    iv$enable()
+    values <- reactive({
+      gather_fails_com(iv)
+      list(
+        "letter" = input[["letter"]],
+        "number" = input[["number"]]
+      )
+    })
+  }
+
+  shiny::testServer(server, {
+    session$setInputs(
+      "letter" = "A",
+      "number" = 2L
+    )
+    expect_identical(values(), list(
+      "letter" = input[["letter"]],
+      "number" = input[["number"]]
+    ))
+  })
+})
+
+test_that("gather_fails_grp: valid inputs produce desired output", {
+  server <- function(input, output, session) {
+    iv <- InputValidator$new()
+    iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter"))
+    iv$add_rule("number", ~ if (as.integer(.) %% 2L == 1L) "choose an even number")
+    iv$enable()
+    values <- reactive({
+      gather_fails_grp(list(iv))
+      list(
+        "letter" = input[["letter"]],
+        "number" = input[["number"]]
+      )
+    })
+  }
+
+  shiny::testServer(server, {
+    session$setInputs(
+      "letter" = "A",
+      "number" = 2L
+    )
+    expect_identical(values(), list(
+      "letter" = input[["letter"]],
+      "number" = input[["number"]]
+    ))
+  })
+})
+
+
+test_that("gather_fails: invalid inputs raise errors in output", {
```

Comment:
I think we should have `testthat::` where needed throughout the file

Reply:
done